### DISCUSSION TITLE
Path filtering and conditional build

### DISCUSSION THREAD

**Discussion by senges** [concourse/concourse#6629] [Help & Support]
Hi, I would like to switch from Github Actions to Concourse-CI but I need to detect file changes in specific folders.

I have 30 folders, with a Dockerfile in each one of them. I don't want to build/push every image at every commit, but only a specific one if  changes have been made.

For example, if my last commit did update a file in "folder-21", I wan't my CI to skip "folder-1", "folder-2" etc.. and only build image for the Dockerfile of "folder-21".

I'm using dorny/paths-filter step in Github Actions, and I would love to find a trick to do the same thing in Concourse.
Thanks !

**Comment by taylorsilva** [comment]
Concourse itself doesn't contain a feature like that, but the resources you use may implement such a feature.
For your situation it sounds like you'd be using the git resource: https://github.com/concourse/git-resource#source-configuration
It has a `path` option which allows you to specify a list of files/folders to only listen to.

**Reply by senges** [reply]
Alright, I'll try to implement a workaround using git-ressource.
Doesn't seems to be very convenient, but now I have a place to start.

Thanks alot !
